src/root.js

```diff
@@ -85,13 +85,13 @@ Root.prototype.load = function load(filename, options, callback) {
 
     // Finishes loading by calling the callback (exactly once)
     function finish(err, root) {
+        if (sync)
+            throw err;
```

Comment:
Doesn't this throw even if there is no error?

Reply:
Yep. But that was true before, too. It works because `finish` is always invoked with a non-null error in synchronous mode.